hbase/src/main/java/org/apache/zeppelin/hbase/HbaseInterpreter.java

```diff
@@ -0,0 +1,148 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.hbase;
+
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterPropertyBuilder;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.jruby.embed.LocalContextScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.jruby.embed.ScriptingContainer;
+
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Support for Hbase Shell. All the commands documented here
+ * https://wiki.apache.org/hadoop/Hbase/Shell is supported.
+ *
+ * Requirements:
+ * Hbase Shell should be installed on the same machine. To be more specific, the following dir.
+ * should be available: https://github.com/apache/hbase/tree/master/hbase-shell/src/main/ruby
+ * Hbase Shell should be able to connect to the Hbase cluster from terminal. This makes sure
+ * that the client is configured properly.
+ *
+ * The interpreter takes 3 config parameters:
+ * hbase.home: Root dir. where hbase is installed. Default is /usr/lib/hbase/
+ * hbase.ruby.sources: Dir where shell ruby code is installed.
+ *                          Path is relative to hbase.home. Default: lib/ruby
+ * hbase.irb.load: (Testing only) Default is true.
+ *                      Whether to load irb in the interpreter.
+ */
+public class HbaseInterpreter extends Interpreter {
+  private Logger logger = LoggerFactory.getLogger(HbaseInterpreter.class);
+  private ScriptingContainer scriptingContainer;
+
+  private StringWriter writer;
+
+  static {
+    Interpreter.register("hbase", "hbase", HbaseInterpreter.class.getName(),
+        new InterpreterPropertyBuilder()
+            .add("hbase.home", "/usr/lib/hbase/", "Installation dir. of Hbase")
+            .add("hbase.ruby.sources", "lib/ruby",
+                "Path to Ruby scripts relative to 'hbase.home'")
+            .add("hbase.irb.load", "true", "Load hirb. Optional for testing only")
+            .build());
+  }
+
+  public HbaseInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    String hbase_home = getProperty("hbase.home");
+    String ruby_src = getProperty("hbase.ruby.sources");
+    String abs_ruby_src = hbase_home + ruby_src;
+
+    logger.info("Home:" + hbase_home);
+    logger.info("Ruby Src:" + ruby_src);
+
+    Properties props = System.getProperties();
```

Comment:
To clarify, I'd suggest adding a check in code, not just logging the value - generally that is done in interpret() method instead of as one could return the error response to display to the user there.